recipe-server/client/control_new/utils/apiFetch.js

```diff
@@ -62,9 +62,9 @@ export default async function apiFetch(url, options = {}) {
     throw new ApiError(message, data);
   }
 
-  if (response.status !== 204) {
+  try {
     return response.json();
```

Comment:
Just tested this locally and we have to change how the `return` works a bit:

```
let json = null;
try {
    json = await response.json();
} catch (error) {}
return json;
```
It seems the `try` wasn't `await`ing for the json call to wrap up before moving on, so when a JSON error comes up after closing a request, the page throws an error and the user isn't redirected. Using the above approach fixed the issue for me.